### ISSUE TITLE
npm run watch is not working

### ISSUE THREAD

**Issue by mbtocalli** [holger1411/understrap#2158]
**Describe the problem**
I have a clean install of understrap
I run "npm install"
Then I run "npm run watch" and i'm getting the following errors:

> understrap@1.2.2 watch
> npm-run-all --parallel watch-run-*

> understrap@1.2.2 watch-run-css
> nodemon --watch src/sass/ --ext scss --exec "npm-run-all css"

> understrap@1.2.2 watch-run-js
> nodemon --watch src/js/ --ext js --exec "npm-run-all js"

[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): src\sass\**\*
[nodemon] watching extensions: scss
[nodemon] starting `npm-run-all css`
[nodemon] 2.0.22
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): src\js\**\*
[nodemon] watching extensions: js
[nodemon] starting `npm-run-all js`

> understrap@1.2.2 css
> npm-run-all css-compile css-postcss css-minify

> understrap@1.2.2 js
> npm-run-all js-compile js-minify

> understrap@1.2.2 css-compile
> sass --style expanded --source-map --embed-sources --no-error-css --quiet src/sass/theme.scss:css/theme.css src/sass/custom-editor-style.scss:css/custom-editor-style.css

> understrap@1.2.2 js-compile
> rollup --config src/build/rollup.config.js --sourcemap

[!] Error: Cannot find module '@rollup/plugin-babel'
Require stack:
- C:\Users\***\Local Sites\***\app\public\wp-content\themes\understrap\src\build\rollup.config.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (C:\Users\Unagui\Local Sites\prueba\app\public\wp-content\themes\understrap\src\build\rollup.config.js:7:19)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at cjsLoader (node:internal/modules/esm/translators:356:17)

ERROR: "js-compile" exited with 1.
ERROR: "js" exited with 1.
[nodemon] app crashed - waiting for file changes before starting...

> understrap@1.2.2 css-postcss
> postcss --config src/build/postcss.config.js --replace "css/*.css" "!css/*.rtl*.css" "!css/*.min.css" "!css/*-bootstrap4.css"

"postcss" is not recognized as an internal or external command,
 operable program or batch file.
ERROR: "css-postcss" exited with 1.
ERROR: "css" exited with 1.
[nodemon] app crashed - waiting for file changes before starting...

**Debug Checklist**
Please include all of the following:
- [ ] The version of Understrap you're using (located in the package.json file)
1.22

- [ ] Whether you are developing with Understrap or the Understrap child theme
Understrap

- [ ] The output of `node -v`
v21.3.0

- [ ] The output of `git --version`
npm git --version:  10.2.4

- [ ] The output of `gulp -v` (if using an Understrap version below 1.0.0)
CLI version 1.3.0

**Comment by bacoords** [comment]
I would try using an older version of node. The recommended version is 16. It's possible that some of the dependencies don't work with 21.

**Comment by mbtocalli** [comment]
Thanks for your responde
I just change to 16.20.2 using nvm
I already tried with 18.16.0 which was the one I used before update to 21.3.0

Still have the same error after npm run watch

Do i have to delete node_modules and reinstall it?

**Comment by bacoords** [comment]
Yes when you change node versions you'll want to delete node_modules and package-lock.json and do a fresh install

**Comment by mbtocalli** [comment]
After changing node version to 16.20.2, deleting node_modules folder and package-lock.json file I do a fresh install using "npm install". This is what I get

npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -4058
npm ERR! enoent An unknown git error occurred
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\***\AppData\Local\npm-cache\_logs\2023-12-04T17_19_38_978Z-debug-0.log

**Comment by pattonwebz** [comment]
How are you getting the repo to work with?

The error message that you see indicates that it can't use `git` so wondering if it's installed on your system and available to call from the terminal that you are running `npm install` from. You can check by typing `git --version` in the terminal.

I just tested with node v16.19.0 and got the installer to work on my side.

**Comment by mbtocalli** [comment]
typing git --version i get:
'git' is not recognized as an internal or external command,
operable program or batch file.

**Comment by pattonwebz** [comment]
That seems to be the issue: ' git` isn't available in the terminal you are running the install. I don't quite understand why `git` is needed for the installation or what package is causing it, but if you can install git in that terminal I think the issue will be solved.

**Comment by mbtocalli** [comment]
I honestly dont understand what is going on.  A few hours ago i was working without a problem. Console suggest me to update caniuse-lite:

Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest

I updated it and after that it doesnt works anymore.

Now i try "npm install git" and i get this:

npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -4058
npm ERR! enoent An unknown git error occurred
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

**Comment by pattonwebz** [comment]
Yeah, it is strange. What operating system are you using, and what terminal are you using to run the `npm install`?

`npm install git` isn't likely to work as git is a system-level software instead of a javascript package.

Knowing what OS and terminal you are using might help us figure out why `git` is not available and how to make it be available there for you.

**Comment by mbtocalli** [comment]
BTW, I solved it simply by deleting the "node_modules" folder and copying the same folder from another local backup installation. After doing that, the command "npm run watch" works as expected so I presume git is already installed at a system-level, right?
Even though I can continue working now, I would like to know how to properly fix this issue.

**Comment by IanDelMar** [comment]
I'm not sure if this is a problem specific to Understrap. Is it reproducible?

**Comment by JamesVanWaza** [comment]
> I'm not sure if this is a problem specific to Understrap. Is it reproducible?

@bacoords and @IanDelMar  I think @mbtocalli has the issue solved, he should test it with `1.2.4` to see if the error is still happening.
